cmd/namespace/sleep_test.go

```diff
@@ -0,0 +1,110 @@
+	"testing"
+
+	"github.com/okteto/okteto/internal/test/client"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/okteto/okteto/pkg/types"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test_SleepNamespace(t *testing.T) {
+	ctx := context.Background()
+	currentNamespace := "current"
+	usr := &types.User{
+		Token: "test-token",
+	}
+	initNamespaces := []types.Namespace{
+		{
+			ID: currentNamespace,
+		},
+		{
+			ID: "test-1",
+		},
+	}
+	var tests = []struct {
+		name string
+		// toSleepNs the namespace to sleep
+		toSleepNs                       string
+		initialNamespacesAtOktetoClient []types.Namespace
+		fakeOkClient                    *client.FakeOktetoClient
+		fakeK8sClient                   *fake.Clientset
+		err                             error
+	}{
+		{
+			name:                            "sleeps existing ns, the current one",
+			toSleepNs:                       currentNamespace,
+			initialNamespacesAtOktetoClient: initNamespaces,
+			fakeOkClient: &client.FakeOktetoClient{
+				Namespace:    client.NewFakeNamespaceClient(initNamespaces, nil),
+				Users:        client.NewFakeUsersClient(usr),
+				StreamClient: client.NewFakeStreamClient(&client.FakeStreamResponse{}),
+			},
+			fakeK8sClient: fake.NewSimpleClientset(),
+		},
+		{
+			name:                            "sleeps existing ns, not the current one",
+			toSleepNs:                       "test-1",
+			initialNamespacesAtOktetoClient: initNamespaces,
+			fakeOkClient: &client.FakeOktetoClient{
+				Namespace:    client.NewFakeNamespaceClient(initNamespaces, nil),
+				Users:        client.NewFakeUsersClient(usr),
+				StreamClient: client.NewFakeStreamClient(&client.FakeStreamResponse{}),
+			},
+			fakeK8sClient: fake.NewSimpleClientset(),
+		},
+		{
+			name:                            "sleep non existing ns",
+			toSleepNs:                       "test-non-existing",
+			initialNamespacesAtOktetoClient: initNamespaces,
+			fakeOkClient: &client.FakeOktetoClient{
+				Namespace:    client.NewFakeNamespaceClient(initNamespaces, nil),
+				Users:        client.NewFakeUsersClient(usr),
+				StreamClient: client.NewFakeStreamClient(&client.FakeStreamResponse{}),
+			},
+			fakeK8sClient: fake.NewSimpleClientset(),
+			err:           errFailedSleepNamespace,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// init ctx current store with initial values
+			okteto.CurrentStore = &okteto.OktetoContextStore{
+				Contexts: map[string]*okteto.OktetoContext{
+					"test-context": {
+						Name:      "test-context",
+						Token:     "test-token",
+						IsOkteto:  true,
+						Namespace: currentNamespace,
+						UserID:    "1",
+					},
+				},
+				CurrentContext: "test-context",
+			}
+			nsFakeCommand := newFakeNamespaceCommand(tt.fakeOkClient, tt.fakeK8sClient, usr)
+			err := nsFakeCommand.ExecuteSleepNamespace(ctx, tt.toSleepNs)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
```

Comment:
Let's use `assert.ErrorIs` We wrap errors and we should check that the error is in the chain